### PR TITLE
[Dataset] Added Prompt Generator,Object Visualizer

### PR DESCRIPTION
**Pull request by archana53**
New PR with handpicked changes from my end.

### CODE REVIEW

File: ovon/dataset/pose_sampler.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from typing import List, Optional, Tuple
```
Comment (Ram81) [inline ovon/dataset/pose_sampler.py:1]: @archana53 can you remove the incomplete files for now? For ex: the `instance_objectnav_generator.py` seems incomplete.

File: ovon/dataset/pose_sampler.py [hunk 2]
```diff
@@ -134,7 +136,7 @@ def sample_agent_poses_radially(self, obj: SemanticObject) -> List[AgentState]:
                         np.deg2rad(deg), habitat_sim.geo.GRAVITY
                     )
 
-                poses.append((AgentState(position=pos, rotation=rot), r))
+                poses.append(AgentState(position=pos, rotation=rot))
```
Comment (Ram81) [inline ovon/dataset/pose_sampler.py:139]: please revert this. I am using the r value for filtering post-hoc

File: ovon/dataset/generate_prompts.py [hunk 3]
```diff
@@ -1,324 +1,357 @@
...
-if os.path.exists(fname_obj):
-    with open(fname_obj, "rb") as f:
-        objects_info = pickle.load(f)
-    
-else:
-    backend_cfg = habitat_sim.SimulatorConfiguration()
-    backend_cfg.scene_id = (
-        scene_path
+
+def isabove(b, a):
```
Comment (Ram81) [inline ovon/dataset/generate_prompts.py:101]: Can you use snakecase for all method and variable names?

File: ovon/dataset/generate_prompts.py [hunk 4]
```diff
@@ -1,324 +1,357 @@
...
+
+    points = np.concatenate(
+        (points_minx, points_maxx, points_miny, points_maxy, points_minz, points_maxz)
     )
-    backend_cfg.scene_dataset_config_file = (
-        "data/scene_datasets/hm3d/hm3d_annotated_basis.scene_dataset_config.json"
+    return points
+
+
+def iscloseto(obj_a, obj_b, delta):
```
Comment (Ram81) [inline ovon/dataset/generate_prompts.py:206]: same here

File: ovon/dataset/instance_objectnav_generator.py [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+from habitat_sim._ext.habitat_sim_bindings import SemanticObject
+from habitat_sim.agent.agent import AgentConfiguration, AgentState, SixDOFPose
+from habitat_sim.simulator import Simulator
+from habitat_sim.utils.common import quat_from_two_vectors, quat_to_coeffs
+from numpy import ndarray
+from ovon.dataset.pose_sampler import PoseSampler
+from ovon.dataset.semantic_utils import ObjectCategoryMapping, get_hm3d_semantic_scenes
+
+# from ovon.dataset.visualization import plot_area  # noqa:F401
+# from ovon.dataset.visualization import save_candidate_imgs
```
Comment (Ram81) [inline ovon/dataset/instance_objectnav_generator.py:25]: remove unused imports

**Review by Ram81 — CHANGES_REQUESTED** [review]

File: ovon/dataset/pose_sampler.py (hunk 1 above)
Comment (archana53) [reply ovon/dataset/pose_sampler.py:1]: Yeah. Got it

**Comment by archana53** [comment]
Made all suggested changes.

File: ovon/dataset/generate_prompts.py [hunk 6]
```diff
@@ -1,324 +1,374 @@
...
+
+    else:
+        with open(
+            f"/nethome/akutumbaka3/files/ovonproject/data/relationships{dim}d_{split}.pickle",
+            "rb",
+        ) as handle:
+            relationships = pickle.load(handle)
+
+    create_html(
+        f"/nethome/akutumbaka3/files/ovonproject/data/webpage/relationships_visualized_{split}.html",
```
Comment (Ram81) [inline ovon/dataset/generate_prompts.py:324]: Can you remove hardcoded paths in follow up PR?

**Review by Ram81 — APPROVED** [review]

**Pull request merged**